pkg/controller/clusterdeployment/clusterdeployment_controller_test.go

```diff
@@ -1443,6 +1444,51 @@ func TestClusterDeploymentReconcile(t *testing.T) {
 				assertConditionReason(t, cd, hivev1.InstallLaunchErrorCondition, "PodInPendingPhase")
 			},
 		},
+		{
+			name: "auth condition when platform creds are bad",
+			existing: []runtime.Object{
+				testClusterDeployment(),
+			},
+			platformCredentialsValidation: func(client.Client, *hivev1.ClusterDeployment, log.FieldLogger) (bool, error) {
+				return false, nil
+			},
+			validate: func(c client.Client, t *testing.T) {
+				cd := getCD(c)
+				require.NotNil(t, cd, "could not get ClusterDeployment")
+
+				assertConditionStatus(t, cd, hivev1.AuthenticationFailureClusterDeploymentCondition, corev1.ConditionTrue)
+			},
+		},
+		{
+			name: "no ClusterProvision when platform creds are bad",
+			existing: []runtime.Object{
+				func() *hivev1.ClusterDeployment {
+					cd := testClusterDeployment()
+					cd.Status.Conditions = []hivev1.ClusterDeploymentCondition{
+						{
+							Status:  corev1.ConditionTrue,
+							Type:    hivev1.AuthenticationFailureClusterDeploymentCondition,
+							Reason:  platformAuthFailureResason,
+							Message: "Platform credentials failed authentication check",
+						},
+					}
+					return cd
+				}(),
+			},
+			platformCredentialsValidation: func(client.Client, *hivev1.ClusterDeployment, log.FieldLogger) (bool, error) {
+				return false, nil
+			},
+			validate: func(c client.Client, t *testing.T) {
+				cd := getCD(c)
+				require.NotNil(t, cd, "could not get ClusterDeployment")
+
+				provisionList := &hivev1.ClusterProvisionList{}
+				err := c.List(context.TODO(), provisionList, client.InNamespace(cd.Namespace))
+				require.NoError(t, err, "unexpected error listing ClusterProvisions")
+
+				assert.Zero(t, len(provisionList.Items), "expected no ClusterProvision objects when platform creds are bad")
+			},
+		},
```

Comment:
I'd like to explore breaking these out into a separate unit test struct if it doesn't result in massive code duplication. If so then this is fine.